src/librustc_ty/ty.rs

```diff
@@ -299,7 +300,23 @@ fn instance_def_size_estimate<'tcx>(
     match instance_def {
         InstanceDef::Item(..) | InstanceDef::DropGlue(..) => {
             let mir = tcx.instance_mir(instance_def);
-            mir.basic_blocks().iter().map(|bb| bb.statements.len()).sum()
+            mir.basic_blocks()
+                .iter()
+                .map(|bb| {
+                    bb.statements
+                        .iter()
+                        .filter(|stmt| match stmt.kind {
+                            StatementKind::StorageLive(_)
+                            | StatementKind::StorageDead(_)
+                            | StatementKind::AscribeUserType(..)
+                            | StatementKind::FakeRead(..)
+                            | StatementKind::Retag(..)
+                            | StatementKind::Nop => false,
+                            _ => true,
+                        })
+                        .count()
```

Comment:
shall there maybe be something like https://github.com/rust-lang/rust/blob/963bf528292d8f97104515e32908e30c2467b6a8/src/librustc_mir/transform/inline.rs#L319-L373
or at least a +1 so the terminator is counted in some way

Reply:
can maybe be simplified to only add extra for landingpads(10 llvm ir instructions) and resume (9 llvm ir instructions)